modules/accounting/includes/functions/products.php

```diff
@@ -139,11 +153,10 @@ function erp_acct_insert_product( $data ) {
         $wpdb->query( 'COMMIT' );
     } catch ( Exception $e ) {
         $wpdb->query( 'ROLLBACK' );
-
-        return new WP_error( 'product-exception', $e->getMessage() );
+        return new WP_Error( 400,  $e->getMessage() );
     }
 
-    return erp_acct_get_product( $product_id );
+    return  $product_id  ;
```

Comment:
Do not change the existing codebase unless necessary, this function might be used by other extensions.